source/patterns/@aws-solutions-constructs/core/lib/glue-job-helper.ts

```diff
@@ -91,7 +98,14 @@ export function buildGlueJob(scope: Construct, props: BuildGlueJobProps): [glue.
         'it is recommended to use "WorkerType" or  "NumberOfWorkers"');
       }
 
-      return deployGlueJob(scope, props.glueJobProps, props.database!, props.table!, props.outputDataStore!);
+      if (!props.glueJobProps.command.scriptLocation && !props.etlCodeAsset) {
```

Comment:
Updated, since in the top level file, we have not ascertained if there is an existing CfnJob instance being passed, I am updating the condition as below

```
    if (!props.existingGlueJob && !props.glueJobProps.command.scriptLocation && !props.etlCodeAsset) {
      throw Error('Either one of CfnJob.JobCommandProperty.scriptLocation or KinesisstreamsToGluejobProps.etlCodeAsset has ' +
      'to be provided. If the ETL Job code file exists in a local filesystem, please set ' +
      'KinesisstreamsToGluejobProps.etlCodeAsset. If the ETL Job is available in an S3 bucket, set the ' +
      'CfnJob.JobCommandProperty.scriptLocation property');
    }
```

So check is valid not CfnJob instance is passed.